### PR TITLE
WIP EXPERIMENTAL: File access in javascript

### PR DESCRIPTION
**Pull request by harshad1**
In this PR I am experimenting with adding a 'Markor' api to javascript opened in the webview. 

This API will allow reading / writing / navigating the file system from javascript in a webview.

May enable useful extensions to markor by users.

### CODE REVIEW

**Comment by harshad1** [comment]
Not sure if we want to merge this at all. Putting it here to drive discussion.

**Comment by gsantner** [comment]
We should not expose file management features to the web/webview, risky to get misused. Some things might be interesting but it needs to be very very careful if it needs to be.

File: app/src/main/java/net/gsantner/markor/web/MarkorWebViewClient.java [hunk 1]
```diff
@@ -62,4 +79,20 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
         }
         return true;
     }
+
+    @Override
+    public void onPageStarted(final WebView view, final String url, final Bitmap favicon) {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/web/MarkorWebViewClient.java:84]: For some added safety I added this to disable file access for non-local files.

Most of this PR is to be looked at as an experiment, not to merged anytime soon :)

**Comment by harshad1** [comment]
Closing this PR. I want to explore other avenues for enabling more functionality from JS

**Comment by gsantner** [comment]
As usual idea and improvements are welcome. Only need to make sure it not obviously poses a viable threat

**Pull request closed without merging**